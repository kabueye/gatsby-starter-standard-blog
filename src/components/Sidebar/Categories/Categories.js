// @flow strict
import React from 'react';
import kebabCase from 'lodash/kebabCase';
import { Link } from 'gatsby';
import styles from './Categories.module.scss';
import { useCategoriesList } from '../../../hooks';

const Categories = () => {
  const categories = useCategoriesList();

  return (<div className={styles['categories']}>
      <div className={styles['categories__inner']}>
        <div className={styles['categories__title']}>Categories</div>
        <ul>
          {categories.map((category) => (
            <li key={category.fieldValue}>
              <Link to={`/category/${kebabCase(category.fieldValue)}/`}>
                {category.fieldValue} ({category.totalCount})
              </Link>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default Categories;
