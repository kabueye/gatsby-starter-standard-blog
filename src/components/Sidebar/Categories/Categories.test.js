// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import Categories from './Categories';
import allMarkdownRemark from '../../../../jest/__fixtures__/all-markdown-remark';
import siteMetadata from '../../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../../types';

describe('Categories', () => {
  const props = {
    ...siteMetadata,
    ...allMarkdownRemark
  };

  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(props)
      ),
      useStaticQuery.mockReturnValue(props)
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Categories />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
