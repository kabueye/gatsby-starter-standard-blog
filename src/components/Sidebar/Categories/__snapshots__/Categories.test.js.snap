// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders correctly 1`] = `
<div
  className="categories"
>
  <div
    className="categories__inner"
  >
    <div
      className="categories__title"
    >
      Categories
    </div>
    <ul>
      <li>
        <a
          href="/category/test-0/"
        >
          test_0
           (
          1
          )
        </a>
      </li>
      <li>
        <a
          href="/category/test-1/"
        >
          test_1
           (
          2
          )
        </a>
      </li>
    </ul>
  </div>
</div>
`;
